crates/bestool/src/actions/crypto/encrypt.rs

```diff
@@ -0,0 +1,63 @@
+use std::{iter, path::PathBuf};
+
+use age::{x25519, Encryptor};
+use clap::Parser;
+use miette::{miette, Context as _, IntoDiagnostic as _, Result};
+use tokio::{fs::File, io::AsyncWriteExt as _};
+use tokio_util::compat::{FuturesAsyncWriteCompatExt as _, TokioAsyncWriteCompatExt as _};
+
+use crate::actions::{
+	crypto::{wrap_async_read_with_progress_bar, CryptoArgs},
+	Context,
+};
+
+#[derive(Debug, Clone, Parser)]
+pub struct EncryptArgs {
+	plaintext: PathBuf,
+
+	#[arg(long)]
+	public_key: PathBuf,
+}
+
+pub async fn run(ctx: Context<CryptoArgs, EncryptArgs>) -> Result<()> {
+	let public_key: x25519::Recipient = tokio::fs::read_to_string(&ctx.args_sub.public_key)
+		.await
+		.into_diagnostic()
+		.wrap_err("reading the public key")?
+		.parse()
+		.map_err(|err: &str| miette!("failed to parse: {err}"))?;
+
+	let plaintext = File::open(&ctx.args_sub.plaintext)
+		.await
+		.into_diagnostic()
+		.wrap_err("opening the plainetxt")?;
+	// Wrap with progress bar before introducing "age" to avoid predicting size after encryption.
+	let mut plaintext = wrap_async_read_with_progress_bar(plaintext).await?;
+
+	let mut encrypted_path = ctx.args_sub.plaintext.into_os_string();
+	encrypted_path.push(".enc");
```

Comment:
It seems the standard is to use `.age` instead of `.enc` [here](https://github.com/FiloSottile/age?tab=readme-ov-file#multiple-recipients)